crates/storage/provider/src/chain.rs

```diff
@@ -217,6 +217,37 @@ impl Chain {
     }
 }
 
+/// Wrapper type for `blocks` display in `Chain`
+pub struct DisplayBlocksChain<'a>(pub &'a BTreeMap<BlockNumber, SealedBlockWithSenders>);
+
+impl<'a> fmt::Display for DisplayBlocksChain<'a> {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        let block_len = self.0.len();
+
+        if block_len == 0 {
+            return Ok(())
+        }
+
+        write!(f, "[{:?}", self.0.iter().next().unwrap().1.num_hash())?;
+        if block_len > 1 {
+            write!(f, ", ")?;
+            for block in self.0.iter().skip(1).take(block_len - 1) {
+                let block_num_hash = block.1.num_hash();
+                write!(
+                    f,
+                    "({:?}, {:?}...), ",
+                    block_num_hash.number,
+                    block_num_hash.hash.to_string().truncate(10)
+                )?;
+            }
+            write!(f, "{:?}", self.0.iter().last().unwrap().1.num_hash())?;
+        }
```

Comment:
what I meant by truncate was truncating all additional entries if len > 3:

`[(first),... (last)]`

Reply:
@mattsse Sorry I misunderstood, I thought you wanted to display only a few characters of the hash in this case. I just modified.